app/sequencescape/sequencescape/api/v2/well.rb

```diff
@@ -107,4 +107,13 @@ def control_info_formatted
   def order_group
     aliquots.map(&:order_group).uniq
   end
+
+  def each_source_barcode_and_location_from_transfer_requests
+    transfer_requests_as_target.each do |transfer_req|
+      # NB. Making assumption here that name field on asset is for a plate well
+      # and contains a plate barcode and well position e.g. DN12345678:A1
+      name_array = transfer_req.source_asset.name.split(':')
+      yield name_array[0], name_array[1] if name_array.length == 2
+    end
+  end
```

Comment:
It's probably safer (albeit slightly slower) to include the labware in the request and pull out the plate barcode explicitly. You can configure includes in the exports.yml file. You might find you need to specify the inclusion of 'labware' rather than plate as (schema wise) you can't be sure if the upstream assets are wells or tubes.

Reply:
I also wonder if this is a bit too specific/view centric to be on well, and might be better as a helper method for the exports controller/views.

Reply:
Oh dear, I've done that thing haven't I. Where a refactor draws attention so something.

Reply:
I moved this to a new helper and extracted more of the common logic as well